lib/streamlit/connections/snowflake_connection.py

```diff
@@ -0,0 +1,217 @@
+# mypy: no-warn-unused-ignores
+
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import TYPE_CHECKING, Sequence, cast
+
+import pandas as pd
+
+from streamlit.connections import BaseConnection
+from streamlit.connections.util import load_from_snowsql_config_file, running_in_sis
+from streamlit.errors import StreamlitAPIException
+from streamlit.runtime.caching import cache_data
+
+if TYPE_CHECKING:
+    from snowflake.connector import (
+        SnowflakeConnection as InternalSnowflakeConnection,  # type: ignore
+    )
+    from snowflake.connector.cursor import SnowflakeCursor  # type: ignore
+    from snowflake.snowpark.session import Session  # type: ignore
+
+
+_REQUIRED_CONNECTION_PARAMS = {"account"}
+
+
+class SnowflakeConnection(BaseConnection["InternalSnowflakeConnection"]):
+    """A connection to Snowflake using the Snowflake Python Connector. Initialize using
+    ``st.connection("<name>", type="snowflake")``.
+
+    TODO(vdonato): Finish this docstring.
+    """
+
+    def _connect(self, **kwargs) -> "InternalSnowflakeConnection":
+        import snowflake.connector  # type: ignore
+        from snowflake.snowpark.context import get_active_session  # type: ignore
+
+        # If we're running in SiS, just call get_active_session() and retrieve the
+        # lower-level connection from it.
+        if running_in_sis():
+            session = get_active_session()
+
+            if hasattr(session, "connection"):
+                return session.connection
+            return session._conn._conn
+
+        # Otherwise, attempt to create a new connection from whatever credentials we
+        # have available.
+
+        st_secrets = self._secrets.to_dict()
+        if len(st_secrets):
+            # TODO(vdonato): Double-check that we want to merge any secrets we find with
+            # the kwargs passed to `st.connection`.
+            conn_kwargs = {**st_secrets, **kwargs}
+            return snowflake.connector.connect(**conn_kwargs)
+
+        if hasattr(snowflake.connector.connection, "CONFIG_MANAGER"):
+            # TODO(vdonato): Double-check that we want to map the "snowflake" connection
+            # name in a call to `st.connection` to the "default" connection in the
+            # Snowflake config file. This is aligned with the default behavior for the
+            # underlying python connector and seems more sane than having a "snowflake"
+            # connection in a snowflake-specific config file.
+            connection_name = (
+                "default"
+                if self._connection_name == "snowflake"
+                else self._connection_name
+            )
+            return snowflake.connector.connect(
+                connection_name=connection_name,
+                **kwargs,
+            )
+
+        # If we have a version of the Snowflake Python Connector installed that predates
+        # the ConfigManager, look for credentials in  ~/.snowsql/config as a last
```

Comment:
With snowcli adoption increasing, check if this is still the (only) path we want to check or not. Don't want to over-complicate this logic though!!

Wonder if we should call out in the docs somewhere "this connection works best with snowflake driver version >= x.x.x" based on when ConfigManager was added

Reply:
> this connection works best with snowflake driver version >= x.x.x

Can you lower pin a dependency to enforce this?

Reply:
@vdonato after thinking about this a bit more, I'm in favor of just removing the approximated snowsql support from the SnowflakeConnection. I think it made sense in the spring, but now that ConfigManager support has been out for 3 months and snowcli workflows are improving, it isn't really needed and it just adds maintenance burden / crufty code.

Problems with what we have now:
- It doesn't actually work for Windows
- It only looks in one of the expected files on Linux, not all of them
- We don't really even document that it works, just a casual mention in the tutorial which suggests that we have the full support (not really true)

We can leave the current implementation in the SnowparkConnection which will be deprecated, but I think we just rip this part out in this updated version.

Then the story is clearer and easier to document explicitly:
- using snowflake-connector-python >= 3.1.0 - you can use Streamlit secrets or the [connections.toml file](https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-example#connecting-using-the-connections-toml-file) which is fully supported and documented, + explicit kwargs
- using an earlier version - you can only use Streamlit secrets or explicit kwargs

WDYT?

Reply:
yep -- this sounds reasonable to me. I'll get rid of the code that attempts to load credentials from the snowsql config file